MetaMorpheus/GuiFunctions/ViewModels/TaskSettingViewModel.cs

```diff
@@ -0,0 +1,354 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+using System.Windows;
+using System.Windows.Input;
+using Easy.Common.Extensions;
+using EngineLayer;
+using Nett;
+using Org.BouncyCastle.Asn1.X509.Qualified;
+using OxyPlot;
+using pepXML.Generated;
+using TaskLayer;
+using ThermoFisher.CommonCore.Data;
+using TopDownProteomics.Chemistry;
+
+namespace GuiFunctions
+{
+    public class TaskSettingViewModel : BaseViewModel
+    {
+        #region Private Properties
+
+        private string selectedSettings;
+        private string typedSettingsName;
+        private Dictionary<string, MetaMorpheusTask> allSettingsDict;
+        private ObservableCollection<string> allSettings => new ObservableCollection<string>(allSettingsDict.Keys);
+
+        #endregion
+
+        #region Public Properties
+
+        public ObservableCollection<string> AllSettings => allSettings;
+
+        public string TypedSettingsName
```

Comment:
Comments should be added to the properties to state what they are used for in the GUI